java/opendcs/src/main/java/decodes/sql/NetworkListListIO.java

```diff
@@ -244,6 +244,112 @@ public void read(NetworkListList nll)
 			nl.addEntry(nle);
 		}
 	}
+
+	/**
+	 Reads the NetworkList and NetworkListEntry tables.
+	 @param nll the list of network lists
+	 @param tmType the transport medium type to filter on
+	 */
+	public void read(NetworkListList nll, String tmType)
+			throws SQLException, DatabaseException
+	{
+		Logger.instance().log(Logger.E_DEBUG1,"Reading NetworkLists...");
+		// Initialize our links into the Database
+		initDb(nll.getDatabase());
+
+		String q = "SELECT * FROM NetworkList";
+
+		if (tmType != null)
+		{
+			String qtmt;
+			if (tmType.contentEquals("goes"))
+			{
+				qtmt = "goes, goes-self-times, goes-random";
+			}
+			else
+			{
+				qtmt = "goes";
+			}
+
+			q = q + " WHERE transportMediumType IN (" + qtmt + ")";
+		}
+
+		Statement stmt = createStatement();
+		ResultSet rs = stmt.executeQuery( q );
+
+		if (rs != null)
+		{
+			while (rs.next())
+			{
+				putNetworkList(DbKey.createDbKey(rs, 1), rs);
+			}
+		}
+
+		// For CWMS, by joining with NetworkList, VPD will automatically
+		// add the predicate to filter by office id. For other DBs, the
+		// join does nothing, but does no harm.
+		String nleColumns = "a.networkListId, a.transportId";
+		if (getDatabaseVersion() >= DecodesDatabaseVersion.DECODES_DB_11)
+			nleColumns += ", a.platform_name, a.description";
+
+		q = "SELECT " + nleColumns
+				+ " FROM NetworkListEntry a, NetworkList b "
+				+ "WHERE a.networkListId = b.id ";
+
+		ResultSet rs_nle = stmt.executeQuery( q );
+
+		while (rs_nle != null && rs_nle.next())
+		{
+			DbKey id = DbKey.createDbKey(rs_nle, 1);
+			String transportId = rs_nle.getString(2);
+
+			NetworkList nl = _networkListList.getById(id);
+			if (nl == null)
+			{
+				Logger.instance().log(Logger.E_WARNING,
+						"Orphan network list entry with invalid network list ID "
+								+ id + ", ignored.");
+				continue;
+			}
+			if (!nl.transportMediumType.equalsIgnoreCase(tmType))
+			{
+				continue;
+			}
+			NetworkListEntry nle = new NetworkListEntry(
+					nl, transportId);
+			if (getDatabaseVersion() >= DecodesDatabaseVersion.DECODES_DB_11)
+			{
+				nle.setPlatformName(rs_nle.getString(3));
+				nle.setDescription(rs_nle.getString(4));
+			}
+			else // Pre version 11 derived name and description from platform & site.
```

Comment:
@MikeNeilson correct me if I'm wrong, but we shouldn't support versions before 11 in new code

Reply:
bit of a judgement call, but I certainly wouldn't hold anyone to doing so.

Reply:
The `DecodesDatabaseVersion` doc says 11 is from OpenDCS 6.1 so wouldn't operating on the latest version of OpenDCS require the schema to be updated?

Reply:
Only to get any of the newest features. We're definitely going to be changing the policy a bit but in the past OpenDCS had always provided support for all previous database schema versions. Hence all the checks you see.

This is very useful for upgrades so we'll be keep backwards compatibly for a few schema version, but yeah, new code especially doesn't need to support anything past the "68" schema.

Reply:
good to know thanks! very generous to support that many versions back